File: GatewayAdapter.java
public class GatewayAdapter implements Payable {

    private IDeal ideal;
    public GatewayAdapter(IDeal ideal) {
        this.ideal = ideal;
    }

    @Override
    public boolean processPayment() {
        return ideal.isValidIban() && ideal.isValidBank() && ideal.isValidAccountName();
    }
}

File: GatewayAdapter.cs
using System;

public class GatewayAdapter : Payable
{
    private IDeal IdealAdapter;

    public GatewayAdapter(IDeal idealAdapter)
    {
        IdealAdapter = idealAdapter;
    }

    public bool ProcessPayment()
    {
        return IdealAdapter.IsValidIban() && IdealAdapter.IsValidBank() && IdealAdapter.IsValidAccountName();
    }
}